### ISSUE TITLE
Regression in 4.5 : missing extensions

### ISSUE THREAD

**Issue by remicollet** [llaville/php-compat-info#210]
Working on mikey179/vfsStream lot of extension are missing in **Extensions Analysis**

     --- 440.txt	2015-10-07 06:25:30.000000000 +0200
     +++ 450.txt	2015-10-07 06:25:59.000000000 +0200
     @@ -6,26 +6,20 @@
      
      Extensions Analysis
     
     -    Extension Matches REF      EXT min/Max     PHP min/Max PHP all 
     -    Core              Core                     4.0.0       5.3.0   
     -    date              date     4.0.0           4.0.0               
     -    dom               dom      5.0.0           5.0.0               
     -    pcre              pcre     4.0.0           4.0.0               
     -    posix             posix    4.0.0           4.0.0               
     -    spl               spl      5.1.0 => 5.2.17 5.1.0       5.2.0   
     -    standard          standard 4.0.0           4.0.0       5.4.0   
     -    xml               xml      4.0.0           4.0.0               
     - C  zip               user                     4.0.0       5.2.0   
     -    Total [9]                                  5.1.0       5.4.0   
     +    Extension Matches REF  EXT min/Max     PHP min/Max PHP all 
     +    Core              Core 5.1.0           5.1.0               
     +    spl               spl  5.1.0 => 5.2.17 5.1.0               
     + C  zip               user                 4.0.0               
     +    Total [3]                              5.1.0               

Ex
* DOMDocument in Classes Analysis is missing
* date (function time) is missing
* pcre (function preg_match) is missing
* posix (function posix), is missing, as conditional
* xml (function xml_parser_create) is missing.

BTW, all "minimum" change from 4.0 to 5.3 which is ok as namespaces used.

**Comment by remicollet** [comment]
There is also a small output between "official" 4.5 and RPM version (where reference DB is re-generated)

      Interfaces Analysis
      
     -    Interface                               Matches REF  EXT min/Max     PHP min/Max PHP all 
     -    Iterator                                1       spl  5.1.0 => 5.2.17 5.1.0               
     +    Interface                               Matches REF  EXT min/Max PHP min/Max PHP all 
     +    Iterator                                1       Core 5.3.0       5.3.0               

(- is RPM version, + is phar version)

I think this is related to sqlite result order as Iterator is both in Spl.interfaces.json (5.1.0/5.2.17) and in Core5.interfaces.json (5.3.0).  I think, 5.1.0 is the correct answer, but probably this should be fixed to be reported correctly, whatever the sqlite result order.

**Comment by llaville** [comment]
I'll have a deeper look later, probably tomorrow evening, but I confirm the regression :(

**Comment by llaville** [comment]
Remi, to help me, could you send me a full report produced with the RPM distrib 4.4.0 and 4.5.1 in text file.
TIP: I used myself the `--output` option like this  

```
php \home\github\php-compat-info\build\phpcompatinfo-4.4.0.phar -vv analyser:run --alias vfs --profile --output \var\logs\phpcompatinfo\vfs\440.txt
```
With `phpcompatinfo.json` declaration such as
```
{
    "source-providers": [
        {
            "in": "phar:///var/dist/vfsStream-1.6.0.zip as vfs",
            "path": "src/",
            "name": "*.php"
        },

```
But you will probably excludes test files from scanning, with 
```
{
    "source-providers": [
        {
            "in": "phar:///var/dist/vfsStream-1.6.0.zip as vfs",
            "path": "src/main/",
            "name": "*.php"
        },

```
Thanks in advance

**Comment by llaville** [comment]
Oh, and I forgot. Please specify which vfsStream distrib you've scanned (1.6.0 or other) and the PHP version used in your platform.

**Comment by remicollet** [comment]
vfsStream 1.6.0 (vfsStream-73bcb605b741a7d5044b47592338c633788b0eb7)
php 5.6.14

**Comment by llaville** [comment]
I've added also the standard vsfStream 1.6 distrib in zip format (see commit https://github.com/llaville/php-compat-info/commit/5667d19b4da8e27f9fd8de96a5a9d0e93d596d61)
License is compatible, and this test may be excluded if necessary: added `@group large`

**Comment by llaville** [comment]
There are perharps other issues, but this one was fixed. Extensions are back !

**Comment by remicollet** [comment]
I confirm 4.5.0 + bcd5f85 and extensions are back.

**Comment by remicollet** [comment]
@llaville do you plan a quick 4.5.1 or should I rebuild with the patch ?

**Comment by llaville** [comment]
Unless you found something wrong again, in the next 24 hour, I plan to release the 4.5.1 tomorrow. 
Is it acceptable for you ?

**Comment by remicollet** [comment]
Yes, perfect.

**Comment by remicollet** [comment]
Notice: my comment https://github.com/llaville/php-compat-info/issues/210#issuecomment-146077967 can worth a look. Nothing really serious, which could probably be mostly fixed by an "ORDER" clause in the sqlite request.

**Comment by llaville** [comment]
Seems working, but we should probably extended it  to all references elements and not such interfaces.
I need more time and regression test before to valid it and include in upcoming 4.5.1
I won't have another regression (stability IS first order ;-)

**Comment by llaville** [comment]
I confirm that your comment about Iterator interface versions will not be included in incoming release 4.5.1
I want to have a better feedback on multiple contexts (use cases).
